src/V3Hasher.cpp

```diff
@@ -29,40 +31,290 @@ class HasherVisitor final : public AstNVisitor {
     // AstUser4InUse     in V3Hasher.h
 
     // STATE
-    V3Hash m_lowerHash;  // Hash of the statement we're building
+    V3Hash m_hash;  // Hash value accumulator
     const bool m_cacheInUser4;  // Use user4 to cache each V3Hash?
 
     // METHODS
     VL_DEBUG_FUNC;  // Declare debug()
 
-    //--------------------
-    virtual void visit(AstVar*) override {}
-    virtual void visit(AstTypedef*) override {}
-    virtual void visit(AstParamTypeDType*) override {}
-
-    virtual void visit(AstNode* nodep) override {
-        V3Hash thisHash;
-        if (!m_cacheInUser4 || !nodep->user4()) {
-            VL_RESTORER(m_lowerHash);
-            {
-                m_lowerHash = nodep->sameHash();
-                UASSERT_OBJ(!m_lowerHash.isIllegal(), nodep,
-                            "sameHash function undefined (returns 0) for node under CFunc.");
-                // For identical nodes, the type should be the same thus
-                // dtypep should be the same too
-                m_lowerHash = V3Hash(m_lowerHash,
-                                     V3Hash(V3Hash(nodep->type() << 6), V3Hash(nodep->dtypep())));
-                // Now update m_lowerHash for our children's (and next children) contributions
-                iterateChildrenConst(nodep);
-                // Store the hash value
-                if (m_cacheInUser4) { nodep->user4(m_lowerHash.fullValue()); }
-                thisHash = m_lowerHash;
-            }
+    inline void walk(AstNode* nodep) {  //
+        if (VL_LIKELY(nodep)) iterate(nodep);
+    }
```

Comment:
Please instead add iterateNull next to the definition for IterateAndNextNull(nodep).